typedData/typedData_test.go

```diff
@@ -0,0 +1,346 @@
+			TypeName:       "Session",
+			ExpectedEncode: `Session(key:felt,expires:felt,root:merkletree)`,
+		},
+		{
+			TypedData:      typedDataExamples["mail_StructArray"],
+			TypeName:       "Mail",
+			ExpectedEncode: `Mail(from:Person,to:Person,posts_len:felt,posts:Post*)Person(name:felt,wallet:felt)Post(title:felt,content:felt)`,
+		},
+		{
+			TypedData:      typedDataExamples["v1Nested"],
+			TypeName:       "TransferERC721",
+			ExpectedEncode: `"TransferERC721"("MessageId":"felt","From":"Account1","To":"Account1","Nft_to_transfer":"Nft","Comment1":"string","Comment2":"string","Comment3":"string")"Account1"("Name":"string","Address":"felt")"Nft"("Collection":"string","Address":"felt","Nft_id":"felt","Negotiated_for":"Transaction")"Transaction"("Qty":"string","Unit":"string","Token_address":"felt","Amount":"felt")`,
+		},
+		{
+			TypedData:      typedDataExamples["example_enum"],
+			TypeName:       "Example",
+			ExpectedEncode: `"Example"("someEnum1":"EnumA","someEnum2":"EnumB")"EnumA"("Variant 1":(),"Variant 2":("u128","u128*"),"Variant 3":("u128"))"EnumB"("Variant 1":(),"Variant 2":("u128"))`,
+		},
+	}
+	for _, test := range testSet {
+		require.Equal(t, test.ExpectedEncode, test.TypedData.types[test.TypeName].EncoddingString)
+	}
+}
+
+// TestGetStructHash tests the GetStructHash function.
+//
+// It creates a mock typed data and calls the GetStructHash method with the
+// type name. It checks if the returned encoding matches the expected
+// encoding. If there is an error during the encoding process, it fails the
+// test.
+//
+// Parameters:
+// - t: The testing.T object used for reporting test failures and logging test output
+// Returns:
+//
+//	none
+func TestGetStructHash(t *testing.T) {
+	type testSetType struct {
+		TypedData    TypedData
+		TypeName     string
+		Context      []string
+		ExpectedHash string
+	}
+	testSet := []testSetType{
+		{
+			TypedData:    typedDataExamples["baseExample"],
+			TypeName:     "StarkNetDomain",
+			ExpectedHash: "0x54833b121883a3e3aebff48ec08a962f5742e5f7b973469c1f8f4f55d470b07",
+		},
+		{
+			TypedData:    typedDataExamples["example_baseTypes"],
+			TypeName:     "Example",
+			ExpectedHash: "0x75db031c1f5bf980cc48f46943b236cb85a95c8f3b3c8203572453075d3d39",
+		},
+		{
+			TypedData:    typedDataExamples["example_presetTypes"],
+			TypeName:     "Example",
+			ExpectedHash: "0x74fba3f77f8a6111a9315bac313bf75ecfa46d1234e0fda60312fb6a6517667",
+		},
+		{
+			TypedData:    typedDataExamples["session_MerkleTree"],
+			TypeName:     "Session",
+			ExpectedHash: "0x73602062421caf6ad2e942253debfad4584bff58930981364dcd378021defe8",
+		},
+		{
+			TypedData:    typedDataExamples["v1Nested"],
+			TypeName:     "TransferERC721",
+			ExpectedHash: "0x11b5fb80dd88c3d8b6239b065def4ac9a79e6995b117ed5940a3a0734324b79",
+		},
+		{
+			TypedData:    typedDataExamples["example_enum"],
+			TypeName:     "Example",
+			ExpectedHash: "0x1551dc992033e2256a2f7ec849495d90f9759ebb535e3006d16e2b9e3b57b4c",
```

Comment:
Where did ExpectedHash come from for these?

Reply:
Explained in my previous comment
https://github.com/NethermindEth/starknet.go/pull/637#discussion_r1879436991